### ISSUE TITLE
Start/End Experiment Date Displayed as Local Time

### ISSUE THREAD

**Issue by dhurlburtusa** [Automattic/abacus#79]
Let's say I am creating an experiment to start 2020-06-01 and end 2020-06-07. Because the `start_datetime` and `end_datetime` are datetime instead of just date fields, we decided to append `T00:00:00.000Z` to the date the user selected. So, `2020-06-01T00:00:00.000Z` is equal to `2020-05-31T18:00:00.000-06:00` in my timezone of MDT. Standard format functions such as `format` from `date-fns` will output `2020-05-31` which can currently be seen in the ExperimentsTable. Note: This issue will only show up in locales west of the prime meridian. Note: The dates correctly come back from the server as `2020-06-01T00:00:00.000+00:00` and `2020-06-07T00:00:00.000+00:00` which we pass to `parseISO` function to get a `Date` instance. Then this `Date` instance is passed to the `format` function.

There are several different ways we can fix this issue.

1) Parse the date without any timezone information. This will cause the date/time to be interpreted as local time at whatever time is in the date string.
2) Similarly, parse the date without any time information or timezone information. This will cause the date/time to be interpreted as local time at midnight.
```
const TIME_AND_TZ_INFO_RE = /T?\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:Z|[+-]\d{2}(?::\d{2})?)?/

function stripTimeAndTzInfo(dateStr: string) {
  return dateStr.replace(TIME_AND_TZ_INFO_RE, '')
}
```
3) Update the `Date` instance by adding the user's timezone offset and then format the date.
4) Just use a date datatype instead of a datetime datatype. The date should then be sent back to the user with the correct date without time or timezone information. This will be similar to option 2 but the client doesn't have to strip the extraneous time info.

**Comment by yanirs** [comment]
We shouldn't assume that the `start/end_datetime`s returned from the server represent round dates. We update them to the actual start/end date + time once the experiment starts/stops. For the table and other places where we _display_ them, we should just display a datetime. This can either be in UTC or in the client's timezone.

For creation/editing, we should only allow round UTC dates for now.

**Comment by dhurlburtusa** [comment]
>We shouldn't assume that the start/end_datetimes returned from the server represent round dates.
Not assuming they are round date. They were round dates in this example. (If by "round dates" you mean 00:00:00Z.)

**Comment by yanirs** [comment]
I believe this was resolved by #118
